Inconsistent ESM adherence causes ERR_MODULE_NOT_FOUND

When I run `npm i solid-swr && node -e 'const {useSwr} = import("solid-swr");'` nodejs throws the following error. I experience this error in node 18, 20, and 22. I believe this is because this package uses `"type": "module"` in package.json, but also ships `.js` as an extension throughout the `dist` directory (instead of `.mjs`). Renaming the appropriate files to .mjs fixes the problem, but this may be undesirable for other reasons.

```
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/home/taylor1791/src/tmp/node_modules/solid-swr/dist/core' imported from /home/taylor1791/src/tmp/node_modules/solid-swr/dist/index.js
    at finalizeResolution (node:internal/modules/esm/resolve:257:11)
    at moduleResolve (node:internal/modules/esm/resolve:913:10)
    at defaultResolve (node:internal/modules/esm/resolve:1037:11)
    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:650:12)
    at #cachedDefaultResolve (node:internal/modules/esm/loader:599:25)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:582:38)
    at ModuleLoader.getModuleJobForImport (node:internal/modules/esm/loader:241:38)
    at ModuleJob._link (node:internal/modules/esm/module_job:132:49) {
  code: 'ERR_MODULE_NOT_FOUND',
  url: 'file:///home/taylor1791/src/tmp/node_modules/solid-swr/dist/core'
}
```

I discovered this problem while I was using vitest. If any tested file imports `solid-swr` the same error occurs since vitest [appears](https://github.com/vitest-dev/vitest/discussions/4233) to follow nodejs' module resolution algorithm.

--- a/package.json
+++ b/package.json
@@ -8,17 +8,14 @@
     "module": "./dist/index.js",
     "exports": {
         ".": {
-            "types": "./dist/index.d.ts",
             "import": "./dist/index.js",
             "require": "./dist/index.cjs"
         },
         "./cache": {
-            "types": "./dist/cache.d.ts",
             "import": "./dist/cache.js",
             "require": "./dist/cache.cjs"
         },
         "./extra": {
-            "types": "./dist/extra.d.ts",
             "import": "./dist/extra.js",
             "require": "./dist/extra.cjs"
         }

--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
     "name": "solid-swr",
     "description": "The beloved swr package ported over to solid",
-    "version": "5.0.1",
+    "version": "5.0.2",
     "type": "module",
     "main": "./dist/index.cjs",
     "types": "./dist/index.d.ts",

--- a/package.json
+++ b/package.json
@@ -4,38 +4,23 @@
     "version": "5.0.1",
     "type": "module",
     "main": "./dist/index.cjs",
-    "types": "./dist/index.d.cts",
+    "types": "./dist/index.d.ts",
     "module": "./dist/index.js",
     "exports": {
         ".": {
-            "import": {
-                "types": "./dist/index.d.ts",
-                "default": "./dist/index.js"
-            },
-            "require": {
-                "types": "./dist/index.d.cts",
-                "default": "./dist/index.cjs"
-            }
+            "types": "./dist/index.d.ts",
+            "import": "./dist/index.js",
+            "require": "./dist/index.cjs"
         },
         "./cache": {
-            "import": {
-                "types": "./dist/cache.d.ts",
-                "default": "./dist/cache.js"
-            },
-            "require": {
-                "types": "./dist/cache.d.cts",
-                "default": "./dist/cache.cjs"
-            }
+            "types": "./dist/cache.d.ts",
+            "import": "./dist/cache.js",
+            "require": "./dist/cache.cjs"
         },
         "./extra": {
-            "import": {
-                "types": "./dist/extra.d.ts",
-                "default": "./dist/extra.js"
-            },
-            "require": {
-                "types": "./dist/extra.d.cts",
-                "default": "./dist/extra.cjs"
-            }
+            "types": "./dist/extra.d.ts",
+            "import": "./dist/extra.js",
+            "require": "./dist/extra.cjs"
         }
     },
     "files": [

--- a/package.json
+++ b/package.json
@@ -3,26 +3,38 @@
     "description": "The beloved swr package ported over to solid",
     "version": "5.0.1",
     "type": "module",
-    "main": "./dist/index.js",
-    "types": "./dist/index.d.ts",
+    "main": "./dist/index.cjs",
+    "types": "./dist/index.d.cts",
     "module": "./dist/index.js",
     "exports": {
         ".": {
             "import": {
                 "types": "./dist/index.d.ts",
                 "default": "./dist/index.js"
+            },
+            "require": {
+                "types": "./dist/index.d.cts",
+                "default": "./dist/index.cjs"
             }
         },
         "./cache": {
             "import": {
                 "types": "./dist/cache.d.ts",
                 "default": "./dist/cache.js"
+            },
+            "require": {
+                "types": "./dist/cache.d.cts",
+                "default": "./dist/cache.cjs"
             }
         },
         "./extra": {
             "import": {
                 "types": "./dist/extra.d.ts",
                 "default": "./dist/extra.js"
+            },
+            "require": {
+                "types": "./dist/extra.d.cts",
+                "default": "./dist/extra.cjs"
             }
         }
     },
@@ -47,7 +59,7 @@
     "license": "MIT",
     "scripts": {
         "dev": "vite",
-        "build": "tsc",
+        "build": "tsup",
         "lint": "eslint . -c ./.eslintrc.cjs",
         "test": "vitest",
         "tsc": "tsc --noEmit"
@@ -55,7 +67,6 @@
     "peerDependencies": {
         "solid-js": "1.x.x"
     },
-    "dependencies": {},
     "devDependencies": {
         "@solidjs/testing-library": "0.8.5",
         "@testing-library/jest-dom": "6.3.0",
@@ -65,6 +76,7 @@
         "jsdom": "24.0.0",
         "prettier": "3.2.4",
         "prettier-config-tronikelis": "3.0.0",
+        "tsup": "8.3.5",
         "typescript": "5.3.3",
         "vite": "5.0.12",
         "vite-plugin-solid": "2.9.1",

--- a/tsup.config.ts
+++ b/tsup.config.ts
@@ -1,0 +1,13 @@
+import { defineConfig } from "tsup";
+
+export default defineConfig({
+    clean: true,
+    target: "esnext",
+    format: ["cjs", "esm"],
+    entry: {
+        extra: "./src/extra.ts",
+        cache: "./src/cache.ts",
+        index: "./src/index.ts",
+    },
+    dts: true,
+});

